yaml/src/main/python/main.py

```diff
@@ -58,6 +45,28 @@ def _get_pipeline_yaml():
         pipeline_yaml = fin.read().decode()
     return pipeline_yaml
 
+def _parse_yaml_pipeline(yaml_pipeline):
```

Comment:
Nit: `_parse_yaml_pipeline` isn't very descriptive, maybe `_extract_jinja_variable_names` would be clearer?